src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,145 @@
+
+# Canopy - Photosynthesis
+@land_compute "photosynthesis_net_canopy" SoilCanopyModel p.canopy.photosynthesis.GPP
+@land_compute "photosynthesis_net_leaf" SoilCanopyModel p.canopy.photosynthesis.An
+@land_compute "respiration_leaf" SoilCanopyModel p.canopy.photosynthesis.Rd
+@land_compute "vcmax25" SoilCanopyModel p.canopy.photosynthesis.Vcmax25
+
+# Canopy - Radiative Transfer
+@land_compute "near_infrared_radiation" SoilCanopyModel p.canopy.radiative_transfer.inc_nir
+@land_compute "near_infrared_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.nir.abs
+@land_compute "near_infrared_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.nir.refl
+@land_compute "near_infrared_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.nir.trans
+@land_compute "photosynthetically_active_radiation" SoilCanopyModel p.canopy.radiative_transfer.inc_par
+@land_compute "photosynthetically_active_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.par.abs
+@land_compute "photosynthetically_active_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.par.refl
+@land_compute "photosynthetically_active_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.par.trans
+@land_compute "radiation_longwave_net" SoilCanopyModel p.canopy.radiative_transfer.LW_n
+@land_compute "radiation_shortwave_net" SoilCanopyModel p.canopy.radiative_transfer.SW_n
+
+## Drivers Module ##
+
+@land_compute "soil_organic_carbon" SoilCanopyModel p.drivers.soc # need to fix this in src/shared_utilities/drivers 
+@land_compute "pressure" SoilCanopyModel p.drivers.P
+@land_compute "rainfall" SoilCanopyModel p.drivers.P_liq
+@land_compute "radiation_longwave_down" SoilCanopyModel p.drivers.LW_d
+@land_compute "radiation_shortwave_down" SoilCanopyModel p.drivers.SW_d
+@land_compute "snowfall" SoilCanopyModel p.drivers.P_snow
+@land_compute "solar_zenith_angle" SoilCanopyModel p.drivers.θs
+@land_compute "specific_humidity" SoilCanopyModel p.drivers.q
+@land_compute "wind_speed" SoilCanopyModel p.drivers.u
+
+## Soil Module ##
+
+@land_compute "infiltration" SoilCanopyModel p.soil.infiltration
+@land_compute "soil_hydraulic_conductivity" SoilCanopyModel p.soil.K
+@land_compute "soil_thermal_conductivity" SoilCanopyModel p.soil.κ
+@land_compute "soil_water_potential" SoilCanopyModel p.soil.ψ
+@land_compute "soil_net_radiation" SoilCanopyModel p.soil.R_n
+@land_compute "soil_temperature" SoilCanopyModel p.soil.T
+
+# Soil - Turbulent Fluxes
+@land_compute "soil_aerodynamic_resistance" SoilCanopyModel p.soil.turbulent_fluxes.r_ae
+@land_compute "soil_latent_heat_flux" SoilCanopyModel p.soil.turbulent_flux.lhf
+@land_compute "soil_sensible_heat_flux" SoilCanopyModel p.soil.turbulent_fluxes.shf
+@land_compute "vapor_flux" SoilCanopyModel p.soil.turbulent_fluxes.vapor_flux
+
+# Soil - SoilCO2
+@land_compute "heterotrophic_respiration" SoilCanopyModel p.soilco2.top_bc
```

Comment:
why is it called top_bc?

Reply:
top boundary condition 

In the equations we calculate CO2 production by microbial respiration at each layer of soil, which produces CO2 (gain), CO2 also diffuses through layers (gain and loss), we calculate CO2 concentration in pores at each layers of soil. 

But most observation of "soil respiration" measure CO2 efflux from the soil, not actual CO2 production by microbes, which can be delayed from production to efflux from surface because of diffusion. In our model diffusion is a function of soil texture (clay soil slower diffusion than sandy soils, for example) and soil moisture (moist soil slower diffusion) and soil temperature. 

So we don't directly calculate soil CO2 efflux in the model, but we can get it from the top boundary condition (that we call top_bc. We have that for all prognostic variables that are calculated by layers in the soil domain)